src/test/scala/repos/jdbc/TableJanitorSpec.scala

```diff
@@ -75,7 +76,7 @@ class TableJanitorSpec extends FlatSpec with MustMatchers with PropertyChecks {
   def genScenario = for {
     state <- genState
     items <- Gen.someOf((state.maxSeen + 1L) to (state.maxSeen + 1000L)).map(_.map(entry(_, 101)))
-  } yield (state, items)
+  } yield (state, items.sortBy(_.pk))
```

Comment:
why add the sortBy?

Reply:
This is a great question! For a bit of context I added the sortBy in this spec because it appears there was a change in the way the ScalaCheck generators work between the old version and the updated version. 

I captured some console println statements from some earlier debugging because I found it very puzzling that the tests were suddenly failing with some library upgrades! In the old version of the ScalaCheck generators, the test items generated had primary keys which were monotonically increasing, like so: 
```
item primary key of 827: state's max seen value 826
item primary key of 829: state's max seen value 827
item primary key of 830: state's max seen value 829
item primary key of 833: state's max seen value 830
item primary key of 834: state's max seen value 833
item primary key of 836: state's max seen value 834
item primary key of 837: state's max seen value 836
```

When I upgraded the testing libraries, I found that the implicit generation of increasing item primary key values was no longer the case. Here's what the same comparison of last seen state to item primary key looks like without the sortBy call in the test:

```
item primary key of 3: state's max seen value 1
item primary key of 2: state's max seen value 1
item primary key of 983: state's max seen value 2
item primary key of 4: state's max seen value 983
item primary key of 2: state's max seen value 1
``` 
Items are no longer in order and therefore fail test assertions. 

So if we add back in the sortBy primary key we get something that looks like our old generated values: 

```
item primary key of 152: state's max seen value 149
item primary key of 154: state's max seen value 152
item primary key of 157: state's max seen value 154
item primary key of 158: state's max seen value 157
item primary key of 159: state's max seen value 158
item primary key of 161: state's max seen value 159
item primary key of 163: state's max seen value 161
item primary key of 165: state's max seen value 163
```
I hope this helps! My effort was primarily to bring the current library in line with the status quo of the old version of the library, but if the more sophisticated scalacheck generators, in fact, reveal a deeper problem, I'm happy to help find a solution.

Reply:
Awesome writeup.  Makes sense to me.